go.mod

```diff
@@ -1,18 +1,18 @@
 module github.com/hashicorp/terraform-provider-corner
 
-go 1.23.7
+go 1.24.4
```

Comment:
Motivation: to use https://pkg.go.dev/testing#T.Context

I think this upgrade is okay since this is a provider and not a Go dependency of other things.

Reply:
Yep that's spot on, the only real reason we've kept all Go modules the exact same version (which ends up with the SDKs driving the Go version of utility providers) is convenience of upgrading all at once 👍🏻